cypress/integration/rancid_tomatillos_tests.js

```diff
@@ -0,0 +1,184 @@
+      cy.intercept({
+        method: 'GET',
+        url: 'https://rancid-tomatillos.herokuapp.com/api/v2/movies/694919'
+      },
+      {
+        statusCode: 404,
+      })
+      .get('div[id="694919"]').click()
+      .get('h2').contains('This is a 400 error message on the Movie Details Card')
+    });
+
+    it('should display an appropriate error message if the network request returns a 500 error', () => {
+      cy.intercept({
+        method: 'GET',
+        url: 'https://rancid-tomatillos.herokuapp.com/api/v2/movies/694919'
+      },
+      {
+        statusCode: 500,
+      })
+      .get('div[id="694919"]').click()
+      .get('h2').contains('This is a 500 error message on the Movie Details Card')
+    });
+
+    it('should provide the user with a way to navigate back to the movie library in the event of an error', () => {
+      cy.intercept({
+        method: 'GET',
+        url: 'https://rancid-tomatillos.herokuapp.com/api/v2/movies/694919'
+      },
+      {
+        statusCode: 500,
+      })
+      .get('div[id="694919"]').click()
+      .get('button[class="back-button"]').click()
+      cy.get('div[class="movie-details-card"]').should('not.exist')
+      cy.get('div[class="movie-container"]').should('be.visible')
+    });
+  });
+
+  describe('Search Bar', () => {
+
+    it('should update rendered movies by title as user types out a search', () => {
+      cy.get('form').type('The')
+      cy.fixture('/data.js').then((data) => {
+
+          const foundMovieTitles = data.movieData.filter(movie => movie.title.includes('The'));
+          foundMovieTitles.forEach(movie => {
+            cy.get('div').contains(movie.title)
+          });
+          cy.get('div[id="539885"]').should('not.exist')
+      });
+    });
+
+    it('should update rendered movies by genre as user types out a search', () => {
+      cy.get('form').type('Action')
+      cy.fixture('/data.js').then((data) => {
+          const searchedMovieTitles = data.movieDetails.filter(movie => movie.genres.includes('Action'));
+          console.log(searchedMovieTitles)
+          searchedMovieTitles.forEach(movie => {
+            cy.get('div').contains(movie.title)
+          });
+          cy.get('div[id="694919"]').should('not.exist')
+      });
+    });
+
+    it('should update rendered movies by overview as user types out a search', () => {
+      cy.get('form').type('thief')
+      cy.fixture('/data.js').then((data) => {
+          const searchedMovieTitles = data.movieDetails.filter(movie => movie.overview.includes('thief'));
+          searchedMovieTitles.forEach(movie => {
+            cy.get('div').contains(movie.title)
+          });
+          cy.get('div[id="152812"]').should('not.exist')
+      });
+    });
+
+    it('should not be visible on Movie Details view', () => {
+      cy.get('div[id="694919"]').click()
+      cy.get('form').should('not.exist')
+    });
+
```

Comment:
The easiest way to handle this search bar visibility issue will be to render the form component in the MovieContainer instead of App.